### PR TITLE
Step name is escaped before it is sent to server (close #160)

### LINKED ISSUE #160: Uncaught error in JSON.parse which brakes test task
Run this test:

```js
"@fixture jj";

"@page http://tema.livejournal.com";

"@test"["qqq"] = {
    '1.Click <b> "tema"': function() {
        var bElement = $("#lj_controlstrip_new").find(":containsExcludeChildren(tema)");
        act.click(bElement, {});
    }
};
```

Test fail will be reported correctly, however after this error occurs in `JSON.parse` within Hammerhead (can't be reproduced in HHM) and task frozes.

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

**Comment by helen-dikareva** [comment]
FPR

File: src/client/runner/runner.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import hammerhead from './deps/hammerhead';
 import testCafeCore from './deps/testcafe-core';
 import RunnerBase from './runner-base';
 import * as browser from '../browser';
+import escapeHtml from 'escape-html';
```
Comment (AlexanderMoskovkin) [inline src/client/runner/runner.js]: move this `import` up. Add external imports before internal

File: test/client/fixtures/runner/runner-test.js [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
 var transport    = testCafeCore.get('./transport');
+var ERROR_TYPE   = testCafeCore.ERROR_TYPE;
 
 var testCafeRunner = window.getTestCafeModule('testCafeRunner');
 var actionBarrier  = testCafeRunner.get('./action-barrier/action-barrier');
 var Runner         = testCafeRunner.get('./runner');
 
 
-transport.batchUpdate                = function (callback) {
+var runner                        = null;
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/runner-test.js]: as I see the `runner` var is not used. You use `testRunner` below. I suggest to rename `testRunner` to `runner`

File: test/client/fixtures/runner/runner-test.js [hunk 3]
```diff
@@ -1,35 +1,63 @@
...
+$.fn.load = function (callback) {
     callback();
 };
 
 Runner.checkStatus = function () {
 };
 
+QUnit.testStart(function () {
+    testRunner                    = new Runner();
+    savedTakeScreenshotOnFails    = SETTINGS.TAKE_SCREENSHOT_ON_FAILS;
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/runner-test.js]: The `SETTINGS.TAKE_SCREENSHOT_ON_FAILS` setting is not changed in the tests. Saving it is redundant

File: test/client/fixtures/runner/runner-test.js [hunk 4]
```diff
@@ -64,32 +88,24 @@ asyncTest('Test iterator should not call Transport.fail twice (without screensho
...
+    var transportFailCount = 0;
 
-    transport.fatalError               = function () {
+    transport.fatalError = function () {
         transportFailCount++;
     };
-    SETTINGS.TAKE_SCREENSHOTS_ON_FAILS = true;
 
-    var testRunner = new Runner();
+    SETTINGS.TAKE_SCREENSHOTS_ON_FAILS = true;
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/runner-test.js]: You have a discrepancy:
`SETTINGS.TAKE_SCREENSHOTS_ON_FAILS` here
`SETTINGS.TAKE_SCREENSHOT_ON_FAILS` above.
Please, correct

**Comment by AlexanderMoskovkin** [comment]
@VasilyStrelyaev please, check the strings

File: src/client/runner/runner.js [hunk 5]
```diff
@@ -132,6 +133,10 @@ Runner.prototype._onTestError = function (err, isAssertion) {
     err.screenshotRequired = SETTINGS.get().TAKE_SCREENSHOTS && SETTINGS.get().TAKE_SCREENSHOTS_ON_FAILS &&
                              this.stepIterator.state.curStepErrors.length < 2;
 
+    // NOTE: we should escape test name because it may contains markup (GH-160)
```
Comment (VasilyStrelyaev) [inline src/client/runner/runner.js]:
```
    // NOTE: we should escape the test name because it may contain markup (GH-160)
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by inikulin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**